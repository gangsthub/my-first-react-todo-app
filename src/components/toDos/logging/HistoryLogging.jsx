// @ts-check

import React, { useEffect } from 'react'
import PropTypes from 'prop-types'

import HistoryItem from './HistoryItem'

const HistoryLogging = ({ todos }) => {
  let el
  useEffect(() => {
    el.scrollTop = el.scrollHeight
  })

  return (
    <div
      className="pa-3 bg-light flex-1 scroll-y mt-3"
      ref={element => {
        el = element
      }}
    >
      {todos.map((todo, i) => (
        <HistoryItem key={i} todo={todo} />
      ))}
    </div>
  )
}

HistoryLogging.propTypes = {
  todos: PropTypes.array
}

export default HistoryLogging
