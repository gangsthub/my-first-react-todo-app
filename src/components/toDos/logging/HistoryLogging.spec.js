import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import HistoryLogging from './HistoryLogging'
import { toDoFactory } from '../todos.state'

const shallowRenderer = new ShallowRenderer()

describe('History', () => {
  let todos
  const text = 'hey'
  beforeEach(() => {
    todos = [toDoFactory({ text }), toDoFactory({ text })].map(todo => ({
      text: todo.text
    }))
  })
  it('prints a list', () => {
    shallowRenderer.render(<HistoryLogging todos={todos} />)
    const output = shallowRenderer.getRenderOutput()
    expect(output).toMatchSnapshot()
    expect(output.props.children.length).toBe(2)
  })
})
