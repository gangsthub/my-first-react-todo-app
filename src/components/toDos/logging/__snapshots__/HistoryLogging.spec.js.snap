// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History prints a list 1`] = `
<div
  className="pa-3 bg-light flex-1 scroll-y mt-3"
>
  <HistoryItem
    todo={
      Object {
        "text": "hey",
      }
    }
  />
  <HistoryItem
    todo={
      Object {
        "text": "hey",
      }
    }
  />
</div>
`;
